### DISCUSSION TITLE
handleDataRequest for API routes

### DISCUSSION THREAD

**Discussion by WesleyYue** [remix-run/remix#5869] [Q&A]
I'm trying to use handleDataRequest in entry.server.ts, but it seems that this function is only triggered on pages that have a default component. For API routes, where there is no default exported component, but rather only action/loader, the function doesn't seem to trigger. Is there a way to get this to trigger for action/loader only routes?

**Comment by lili21** [comment]
Could you provide a repo to reproduce this problem? or share some code? it should work for resource routes.
